Add OCP 4.20 jobs

vkmc opened the pull request:
Add job definitions using crc-cloud-ocp-4-20-1-3xl as base image

Closes: OSPRH-21881

vkmc committed:
Add OCP 4.20 jobs

Add job definitions using crc-cloud-ocp-4-20-1-3xl as base image

Closes: OSPRH-21881

vkmc commented:
recheck

vkmc commented:
The label selected for the CRC image is listed in https://softwarefactory-project.io/zuul/t/rdoproject.org/labels. For some reason CRC fails to start. Taking https://github.com/openstack-k8s-operators/ci-framework/pull/3517 as reference.

elfiesmelfie commented on .zuul.yaml:
```diff
@@ -15,6 +15,14 @@
         label: crc-cloud-ocp-4-18-1-3xl
 
+- nodeset:
+    name: stf-crc_extracted-ocp420
```
nit: we should use "cloud" instead of "extracted" in the nodeset name.

elfiesmelfie reviewed (commented):
Minor update for the nodeset name

vkmc commented:
recheck

vkmc committed:
Use OCP 4.20 ci-framework defined nodeset

Directly use centos-9-medium-crc-cloud-ocp-4-20-1-3xl
instead of defining our own nodeset

https://github.com/openstack-k8s-operators/ci-framework/blob/main/zuul.d/nodeset.yaml

vkmc commented:
recheck

vkmc committed:
Revert "Use OCP 4.20 ci-framework defined nodeset"

This reverts commit 7b75b19a599f0d6df7bd2836c2c9ecd5ac2235ff.

vkmc committed:
Test label crc-cloud-ocp-4-20-1-xxl

vkmc committed:
Revert "Test label crc-cloud-ocp-4-20-1-xxl"

This reverts commit d65e61dade200ec6753914712f16a8f73eee185d.

danpawlik commented:
recheck

danpawlik commented:
recheck

vkmc committed:
Update nodesets name to stf-crc_cloud-ocpX.Y

We no longer use crc extracted images, we now use crc cloud images

Update the naming accordingly

vkmc commented:
Running only local build jobs to make it consume less resources while we figure out what is missing from the OCP 4.20 jobs.

vkmc commented:
recheck

vkmc commented:
Deployed an OCP 4.20.5 environment with STF using local builds (as the job currently being inspected). The environment took a bit longer than usual to come up, but all the services reconciled and got to a healthy state.

In the job, though, I can see it is failing early with the following error

```
Uploading directory "/home/zuul/src/github.com/infrawatch/service-telemetry-operator" as binary input for the build ...

Uploading finished
Error from server (InternalError): Internal error occurred: error dialing backend: remote error: tls: internal error
```

vkmc commented:
recheck

vkmc committed:
Merge branch 'master' into OSPRH-21881/ocp4.20-job-zuul

vkmc committed:
Merge branch 'master' into OSPRH-21881/ocp4.20-job-zuul

vkmc committed:
Drop OCP 4.16 jobs

vkmc committed:
Enable deploy from catalog job

vkmc commented:
recheck

vkmc committed:
[DNM] Add debug steps to get more info of the env

vkmc committed:
Raise the waiting time in validate_deployment.sh

Wait for 5 minutes instead of 3 seconds to make sure
that the environment is stable

vkmc committed:
[DNM] Get cert-manager operator debug info

vkmc commented:
recheck

vkmc commented:
recheck

vkmc commented:
recheck

vkmc commented:
recheck

vkmc commented:
recheck

vkmc commented:
recheck

vkmc committed:
Enable all jobs for OCP 4.18 and OCP 4.20

vkmc committed:
Revert "[DNM] Get cert-manager operator debug info"

This reverts commit bcb5cda022d15846923e6a4da7674ac36469b642.

vkmc committed:
Revert "Raise the waiting time in validate_deployment.sh"

This reverts commit 3eef2b73e4f47c90a6bce6a8b87d5fbaffc39c14.

vkmc committed:
Revert "[DNM] Add debug steps to get more info of the env"

This reverts commit a4b1998f6a336fe05f435667ecf382f728e67252.

vkmc commented:
recheck

vkmc commented:
recheck

vkmc commented:
recheck
